Makefile

```diff
@@ -0,0 +1,112 @@
+.PHONY: test-coverage clean install dev format lint all server build upload-test upload release deptry mypy test-mcp test-mcp-extended test-integration test-unit test-real-api
+
+# Default target
+all: clean install dev test-coverage format lint mypy deptry build test-mcp test-mcp-extended test-integration
+
+# Install everything for development
+dev:
+	uv sync --group dev
+
+# Install production only
+install:
+	uv sync
+
+# Run tests with coverage
+test-coverage:
+	uv run pytest --cov=nmdc_mcp --cov-report=html --cov-report=term tests/
+
+# Clean up build artifacts
+clean:
+	rm -rf build/
+	rm -rf dist/
+	rm -rf *.egg-info
+	rm -rf htmlcov/
+	rm -f .coverage
+	find . -type d -name __pycache__ -exec rm -rf {} +
+	find . -type f -name "*.pyc" -delete
+	rm -rf src/*.egg-info
+
+# Run server mode
+server:
+	uv run python src/nmdc_mcp/main.py
+
+# Format code with black
+format:
+	uv run black src/ tests/
+
+lint:
+	uv run ruff check --fix src/ tests/
+
+# Check for unused dependencies
+deptry:
+	uvx deptry .
+
+# Type checking
+mypy:
+	uv run mypy src/
+
+# Build package with uv
+build:
+	uv build
+
+# Upload to TestPyPI (using token-based auth)
+upload-test:
+	uv run twine upload --repository testpypi dist/*
+
+# Upload to PyPI (using token-based auth - set TWINE_PASSWORD environment variable first)
+upload:
+	uv run twine upload dist/*
```

Comment:
I see, so we need the `twine` dependency to upload the package to TestPyPI? curious if can we do it without twine, like how we do it for PyPI?

Not high priority, just a note for a possible refactor PR.

Reply:
I wanted to keep the PyPI publishing as similar as possible to what you already had, but I see this slipped though the cracks.

Reply:
https://github.com/microbiomedata/nmdc-mcp/issues/15